server-channel.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+type ServerChannel interface {
+	Copy(src string, dst string) error
+	Exec(command ...string) error
+	Close()
+}
+
+type CommandRunner func(command string, args ...string) error
+
+// Setup port forwarding, and
+type SshServerChannel struct {
+	commandRunner CommandRunner
+	cmd           *exec.Cmd
```

Comment:
This should probably be called something like tunnel, and could just be the Process.